docs/react-error-boundaries.md

```diff
@@ -0,0 +1,84 @@
+# How to use react error boundaries in fable
+
+When react renders the view and encounters an error it will by default just crash the complete application.
+If your html page only contains a single react element your page will be essentially a white page from this point and the user needs to reload the website.
+
+The problem is that a simple `try-catch` in the view will not be enough to handle errors in the render-pipeline. 
+In order to catch these errors you need to overwrite `componentDidCatch`, this process is documented [here](https://reactjs.org/docs/error-boundaries.html)
+
+While render-errors are rare when using fable it can happen, especially when embedding 3rd-party-components.
+Because of this we provide a general purpose ReactErrorBoundary component which can be added and used from your application (including elmish-architecture)
+
+
+## ReactErrorBoundaries.fs
+
+Just copy the following code to your project:
```

Comment:
I would be in favor of adding paket instruction too. What do you think ?

Reply:
I guess you mean adding the file as loose file to this repository?

Reply:
Thanks a lot for this @matthid! This a great addition to the documentation. Yes, it may be a good idea to add this code to the library to make it easier for users. If you want we can merge this and I'll do it myself.

Reply:
I'm happy either way. I'd assume it is faster if you do it but I'm happy to help (I'm just not exactly sure if you are now suggesting to just add the file or if you suggest to add it to the react library)

Reply:
In fact if we add it to the library I'd suggest fixing the binding for the second parameter ...

Reply:
Ok, let's merge then and I'll try to add it to the library. Thanks again for your help!